src/index.ts

```diff
@@ -142,35 +142,36 @@ export function either<T>(parser: Parser<T>): Parser<{isError: boolean, value: T
   });
 };
 
-// coroutine :: (() -> Iterator (Parser e a s)) -> Parser e a s
-export function coroutine<T>(g: FnReturingParserIterator<T>): Parser<T> {
+// coroutine :: (() -> e) -> Parser e a s
+export function coroutine<T>(scanner: (tokenizer:<K>(parser:Parser<K>)=>K)=>T): Parser<T> {
```

Comment:
The terms `tokenizer` and `scanner` are a bit overloaded - let's use something else, like `parserFn` for scanner, and `yield` for tokenizer. The user can actually name yield whatever they like, but this way it's a little closer to what it was before, so upgrading users will have an easier time.

Reply:
Usage of `yield` as variable name is not allowed strict mode.
can you suggest a new name

Reply:
ah good point. Maybe `parseValue`?

Reply:
Or `run`, which is closer to the `Parser.run` method usage.

Reply:
I named it `_yield` which is more understandable

Reply:
Let's avoid names starting with underscores as it's more awkward to type, and typically refers to either private or unused variables.

Reply:
I'd also like to move away from scanner, since it's a term that's already used in parsing, but in a different context (lexing tokens character by character). That can cause unnecessary confusion. Something like `parserFn` would be better IMO.

Reply:
I used `parserFn` and `run` as names and updated coresponding places